sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -396,9 +412,10 @@ public virtual Task<string[]> GetPartitionIdsAsync(CancellationToken cancellatio
                                               SendEventOptions options,
                                               CancellationToken cancellationToken = default)
         {
+            Argument.AssertNotNull(events, nameof(events));
+
             options ??= DefaultSendOptions;
```

Comment:
nit:  I'd like to keep the assertions together after we assign the defaults.    That's the pattern that we've been following.     

- Setup any defaults that we can
- Validate that we've got the stuff we need to take action
- Do things
- ...
- Profit!

Reply:
Good point, I was thinking about that as well actually.

In hindsight, the problem I faced was the ```NullReferenceException``` triggered from ```AssertSinglePartitionReference(options?.PartitionId, options?.PartitionKey);``` when the ```options``` parameter is passed ```null```.

Was thinking of two possible ways to skin the cat:
- we treat ```options ??= DefaultSendOptions;``` as if it was a means to default parameters and we move it before the assertions
- we prevent the ```NullReferenceException``` from happening ```AssertSinglePartitionReference(options?.PartitionId, options?.PartitionKey);```

Look forward to knowing your thinking on this.

Reply:
I went with the second for now, happy to revisit it as you may find appropriate.

Reply:
We typically do the defaults before validating, so in your case:

```csharp
public Task MethodThing(SendEventOptions options)
{
    options ??= DefaltSendOptions;

    Argument.AssertNotNull(events, nameof(events);
    AssertSinglePartitionReference(options.PartitionId, options.PartitionKey);

    // .... more stuffs
}
```